### ISSUE TITLE
Audio Player doesn't work

### ISSUE THREAD

**Issue by palapapa** [KristofferStrube/Blazor.WebAudio#5]
When trying to use the Audio Player on your website, clicking the play button gives the following error:
```
05:49:21.969
crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: {"name":"TypeError","message":"BaseAudioContext.decodeAudioData: Argument 2 is not an object.","stack":"decodeAudioData@https://kristofferstrube.github.io/Blazor.WebAudio/_content/KristofferStrube.Blazor.WebAudio/KristofferStrube.Blazor.WebAudio.js:92:25\ncallAsyncInstanceMethod@https://kristofferstrube.github.io/Blazor.WebAudio/_content/KristofferStrube.Blazor.WebIDL/KristofferStrube.Blazor.WebIDL.js:131:39\nbeginInvokeJSFromDotNet/i<@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/blazor.webassembly.js:1:2878\nbeginInvokeJSFromDotNet@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/blazor.webassembly.js:1:2835\nvn@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/blazor.webassembly.js:1:58849\nTl</S<@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/dotnet.runtime.8.0.6.ft3vm0pt8p.js:3:178365\nTl<@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/dotnet.runtime.8.0.6.ft3vm0pt8p.js:3:179198\n@:wasm-function[8]:0x0\n@:wasm-function[37965]:0xbea2cc\n@:wasm-function[38044]:0xbef4c1\n@:wasm-function[59638]:0x109676b\n@:wasm-function[71802]:0x1409a18\n@:wasm-function[71963]:0x1426a70\n@:wasm-function[71700]:0x140638a\n@:wasm-function[71692]:0x13f9787\n@:wasm-function[71760]:0x1408902\n@:wasm-function[71768]:0x1408bfd\n@:wasm-function[63925]:0x11b86a5\n@:wasm-function[15654]:0x4b3fc5\n@:wasm-function[15761]:0x4bf297\n@:wasm-function[15640]:0x4b2e62\n@:wasm-function[10608]:0x346c80\n@:wasm-function[54114]:0xf3f9ff\n@:wasm-function[54851]:0xf62b64\n@:wasm-function[50009]:0xe64caf\n@:wasm-function[54120]:0xf3fdee\n@:wasm-function[54119]:0xf3fc81\n@:wasm-function[55307]:0xf7ab01\n@:wasm-function[71802]:0x1409a05\n@:wasm-function[71963]:0x1426a70\n@:wasm-function[71700]:0x140638a\n@:wasm-function[71692]:0x13f96ae\n@:wasm-function[71760]:0x1408902\n@:wasm-function[71783]:0x14091c4\n@:wasm-function[58125]:0x103626c\n@:wasm-function[50751]:0xe89d91\n@:wasm-function[50749]:0xe89244\n@:wasm-function[50619]:0xe8209c\n@:wasm-function[50615]:0xe81eca\n@:wasm-function[51376]:0xea0fb3\n@:wasm-function[51375]:0xea0ee0\n@:wasm-function[10677]:0x34af24\n@:wasm-function[54114]:0xf3f9ff\n@:wasm-function[54851]:0xf62b64\n@:wasm-function[50009]:0xe64caf\n@:wasm-function[54120]:0xf3fdee\n@:wasm-function[54119]:0xf3fc81\n@:wasm-function[55307]:0xf7ab01\n@:wasm-function[71802]:0x1409a05\n@:wasm-function[71963]:0x1426a70\n@:wasm-function[71700]:0x140638a\n@:wasm-function[71692]:0x13f96ae\n@:wasm-function[71760]:0x1408902\n@:wasm-function[71783]:0x14091c4\n@:wasm-function[58125]:0x103626c\n@:wasm-function[50751]:0xe89d91\n@:wasm-function[50749]:0xe89244\n@:wasm-function[50619]:0xe8209c\n@:wasm-function[50615]:0xe81eca\n@:wasm-function[51376]:0xea0fb3\n@:wasm-function[51402]:0xea1a26\n@:wasm-function[15670]:0x4b6184\n@:wasm-function[54114]:0xf3f9ff\n@:wasm-function[54851]:0xf62b64\n@:wasm-function[50009]:0xe64caf\n@:wasm-function[54120]:0xf3fdee\n@:wasm-function[54119]:0xf3fc81\n@:wasm-function[55307]:0xf7ab01\n@:wasm-function[71802]:0x1409a05\n@:wasm-function[71963]:0x1426a70\n@:wasm-function[71700]:0x140638a\n@:wasm-function[71692]:0x13f96ae\n@:wasm-function[71760]:0x1408902\n@:wasm-function[71783]:0x14091c4\n@:wasm-function[58125]:0x103626c\n@:wasm-function[50751]:0xe89d91\n@:wasm-function[50749]:0xe89244\n@:wasm-function[50619]:0xe8209c\n@:wasm-function[50615]:0xe81eca\n@:wasm-function[50863]:0xe91130\n@:wasm-function[50862]:0xe91099\n@:wasm-function[15746]:0x4be846\n@:wasm-function[15661]:0x4b4ba6\n@:wasm-function[15659]:0x4b49ca\n@:wasm-function[15695]:0x4ba0c2\n@:wasm-function[14297]:0x457694\n@:wasm-function[54546]:0xf53374\n@:wasm-function[14282]:0x45580b\n@:wasm-function[14281]:0x455702\n@:wasm-function[14290]:0x4570fe\n@:wasm-function[62740]:0x11629be\n@:wasm-function[71802]:0x1409a05\n@:wasm-function[71963]:0x1426a70\n@:wasm-function[71700]:0x140638a\n@:wasm-function[71692]:0x13f9787\n@:wasm-function[71726]:0x1407227\n@:wasm-function[74414]:0x14bcc76\n@:wasm-function[73784]:0x1491e22\n@:wasm-function[73790]:0x149278a\n@:wasm-function[73814]:0x1494f84\n@:wasm-function[79522]:0x15e4107\ncreateDotnetRuntime/Module._mono_wasm_invoke_method_bound@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/dotnet.native.8.0.6.pa7gdnl2df.js:8:138068\nkr@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/dotnet.runtime.8.0.6.ft3vm0pt8p.js:3:35365\nTl</U<@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/dotnet.runtime.8.0.6.ft3vm0pt8p.js:3:180322\nendInvokeJSFromDotNet@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/blazor.webassembly.js:1:45213\nbeginInvokeJSFromDotNet/<@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/blazor.webassembly.js:1:2967\n"}
      callAsyncInstanceMethod@https://kristofferstrube.github.io/Blazor.WebAudio/_content/KristofferStrube.Blazor.WebIDL/KristofferStrube.Blazor.WebIDL.js:134:15
      
Microsoft.JSInterop.JSException: {"name":"TypeError","message":"BaseAudioContext.decodeAudioData: Argument 2 is not an object.","stack":"decodeAudioData@https://kristofferstrube.github.io/Blazor.WebAudio/_content/KristofferStrube.Blazor.WebAudio/KristofferStrube.Blazor.WebAudio.js:92:25\ncallAsyncInstanceMethod@https://kristofferstrube.github.io/Blazor.WebAudio/_content/KristofferStrube.Blazor.WebIDL/KristofferStrube.Blazor.WebIDL.js:131:39\nbeginInvokeJSFromDotNet/i<@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/blazor.webassembly.js:1:2878\nbeginInvokeJSFromDotNet@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/blazor.webassembly.js:1:2835\nvn@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/blazor.webassembly.js:1:58849\nTl</S<@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/dotnet.runtime.8.0.6.ft3vm0pt8p.js:3:178365\nTl<@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/dotnet.runtime.8.0.6.ft3vm0pt8p.js:3:179198\n@:wasm-function[8]:0x0\n@:wasm-function[37965]:0xbea2cc\n@:wasm-function[38044]:0xbef4c1\n@:wasm-function[59638]:0x109676b\n@:wasm-function[71802]:0x1409a18\n@:wasm-function[71963]:0x1426a70\n@:wasm-function[71700]:0x140638a\n@:wasm-function[71692]:0x13f9787\n@:wasm-function[71760]:0x1408902\n@:wasm-function[71768]:0x1408bfd\n@:wasm-function[63925]:0x11b86a5\n@:wasm-function[15654]:0x4b3fc5\n@:wasm-function[15761]:0x4bf297\n@:wasm-function[15640]:0x4b2e62\n@:wasm-function[10608]:0x346c80\n@:wasm-function[54114]:0xf3f9ff\n@:wasm-function[54851]:0xf62b64\n@:wasm-function[50009]:0xe64caf\n@:wasm-function[54120]:0xf3fdee\n@:wasm-function[54119]:0xf3fc81\n@:wasm-function[55307]:0xf7ab01\n@:wasm-function[71802]:0x1409a05\n@:wasm-function[71963]:0x1426a70\n@:wasm-function[71700]:0x140638a\n@:wasm-function[71692]:0x13f96ae\n@:wasm-function[71760]:0x1408902\n@:wasm-function[71783]:0x14091c4\n@:wasm-function[58125]:0x103626c\n@:wasm-function[50751]:0xe89d91\n@:wasm-function[50749]:0xe89244\n@:wasm-function[50619]:0xe8209c\n@:wasm-function[50615]:0xe81eca\n@:wasm-function[51376]:0xea0fb3\n@:wasm-function[51375]:0xea0ee0\n@:wasm-function[10677]:0x34af24\n@:wasm-function[54114]:0xf3f9ff\n@:wasm-function[54851]:0xf62b64\n@:wasm-function[50009]:0xe64caf\n@:wasm-function[54120]:0xf3fdee\n@:wasm-function[54119]:0xf3fc81\n@:wasm-function[55307]:0xf7ab01\n@:wasm-function[71802]:0x1409a05\n@:wasm-function[71963]:0x1426a70\n@:wasm-function[71700]:0x140638a\n@:wasm-function[71692]:0x13f96ae\n@:wasm-function[71760]:0x1408902\n@:wasm-function[71783]:0x14091c4\n@:wasm-function[58125]:0x103626c\n@:wasm-function[50751]:0xe89d91\n@:wasm-function[50749]:0xe89244\n@:wasm-function[50619]:0xe8209c\n@:wasm-function[50615]:0xe81eca\n@:wasm-function[51376]:0xea0fb3\n@:wasm-function[51402]:0xea1a26\n@:wasm-function[15670]:0x4b6184\n@:wasm-function[54114]:0xf3f9ff\n@:wasm-function[54851]:0xf62b64\n@:wasm-function[50009]:0xe64caf\n@:wasm-function[54120]:0xf3fdee\n@:wasm-function[54119]:0xf3fc81\n@:wasm-function[55307]:0xf7ab01\n@:wasm-function[71802]:0x1409a05\n@:wasm-function[71963]:0x1426a70\n@:wasm-function[71700]:0x140638a\n@:wasm-function[71692]:0x13f96ae\n@:wasm-function[71760]:0x1408902\n@:wasm-function[71783]:0x14091c4\n@:wasm-function[58125]:0x103626c\n@:wasm-function[50751]:0xe89d91\n@:wasm-function[50749]:0xe89244\n@:wasm-function[50619]:0xe8209c\n@:wasm-function[50615]:0xe81eca\n@:wasm-function[50863]:0xe91130\n@:wasm-function[50862]:0xe91099\n@:wasm-function[15746]:0x4be846\n@:wasm-function[15661]:0x4b4ba6\n@:wasm-function[15659]:0x4b49ca\n@:wasm-function[15695]:0x4ba0c2\n@:wasm-function[14297]:0x457694\n@:wasm-function[54546]:0xf53374\n@:wasm-function[14282]:0x45580b\n@:wasm-function[14281]:0x455702\n@:wasm-function[14290]:0x4570fe\n@:wasm-function[62740]:0x11629be\n@:wasm-function[71802]:0x1409a05\n@:wasm-function[71963]:0x1426a70\n@:wasm-function[71700]:0x140638a\n@:wasm-function[71692]:0x13f9787\n@:wasm-function[71726]:0x1407227\n@:wasm-function[74414]:0x14bcc76\n@:wasm-function[73784]:0x1491e22\n@:wasm-function[73790]:0x149278a\n@:wasm-function[73814]:0x1494f84\n@:wasm-function[79522]:0x15e4107\ncreateDotnetRuntime/Module._mono_wasm_invoke_method_bound@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/dotnet.native.8.0.6.pa7gdnl2df.js:8:138068\nkr@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/dotnet.runtime.8.0.6.ft3vm0pt8p.js:3:35365\nTl</U<@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/dotnet.runtime.8.0.6.ft3vm0pt8p.js:3:180322\nendInvokeJSFromDotNet@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/blazor.webassembly.js:1:45213\nbeginInvokeJSFromDotNet/<@https://kristofferstrube.github.io/Blazor.WebAudio/_framework/blazor.webassembly.js:1:2967\n"}
callAsyncInstanceMethod@https://kristofferstrube.github.io/Blazor.WebAudio/_content/KristofferStrube.Blazor.WebIDL/KristofferStrube.Blazor.WebIDL.js:134:15

   at KristofferStrube.Blazor.WebIDL.ErrorHandlingJSObjectReference.<InvokeAsync>d__9`1[[Microsoft.JSInterop.IJSObjectReference, Microsoft.JSInterop, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].MoveNext()
   at KristofferStrube.Blazor.WebIDL.ErrorHandlingJSObjectReference.<InvokeAsync>d__8`1[[Microsoft.JSInterop.IJSObjectReference, Microsoft.JSInterop, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].MoveNext()
   at KristofferStrube.Blazor.WebAudio.BaseAudioContext.DecodeAudioDataAsync(IJSObjectReference audioData, Func`2 successCallback, Func`2 errorCallback)
   at KristofferStrube.Blazor.WebAudio.BaseAudioContext.DecodeAudioDataAsync(Byte[] audioData, Func`2 successCallback, Func`2 errorCallback)
   at KristofferStrube.Blazor.WebAudio.BaseAudioContext.DecodeAudioDataAsync(Byte[] audioData, Func`2 successCallback, Func`2 errorCallback)
   at KristofferStrube.Blazor.WebAudio.WasmExample.Pages.AudioPlayer.EnsureCurrentTrackLoaded()
   at KristofferStrube.Blazor.WebAudio.WasmExample.Pages.AudioPlayer.Play()
   at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task , ComponentState )
blazor.webassembly.js:1:46958

```

**Comment by KristofferStrube** [comment]
Hey @palapapa I don't see this problem. Could you share details about what browser and version you are using?

**Comment by palapapa** [comment]
Firefox 128.0.3

**Comment by KristofferStrube** [comment]
I have reproduced it now. It seems that there is some difference in how Firefox and Chrome have implemented the API.

Chrome allows for us to parse null for parameters that are optional and Firefox does not seem to support this.

I have pushed a fix for that part now and will make a new alpha release soon with that change.

While I looked into this problem I also found that error handling JSInterop did not seem to work in Firefox. I will try to find out if this is a general problem for Firefox or if this is only the case for this method.
